yt_dlp/extractor/bilibili.py

```diff
@@ -487,16 +502,43 @@ def _real_extract(self, url):
         webpage = self._download_webpage(url, media_id)
 
         initial_state = self._search_json(r'window\.__INITIAL_STATE__\s*=', webpage, 'initial_state', media_id)
-        episode_list = self._download_json(
-            'https://api.bilibili.com/pgc/web/season/section', media_id,
-            query={'season_id': initial_state['mediaInfo']['season_id']},
-            note='Downloading season info')['result']['main_section']['episodes']
+        episode_list = traverse_obj(
+            self._download_json('https://api.bilibili.com/pgc/web/season/section', media_id,
+                                query={'season_id': traverse_obj(initial_state, ('mediaInfo', 'season_id'))},
+                                headers={'Referer': url, **self.geo_verification_headers()},
+                                note='Downloading season info'),
+            ('result', 'main_section', 'episodes'))
```

Comment:
we can filter to force a list of useful results
```suggestion
            ('result', 'main_section', 'episodes', lambda _, v: url_or_none(v['share_url']) and v['id']))
```